weaknesses/MASVS-RESILIENCE/MASWE-0098.md

```diff
@@ -2,17 +2,43 @@
 title: App Virtualization Environment Detection Not Implemented
 id: MASWE-0098
 alias: app-virtualization-detection
-platform: [android, ios]
+platform: [android]
 profiles: [R]
 mappings:
-  masvs-v2: [MASVS-RESILIENCE-1]
-
-draft:
-  description: runs as a so-called "clone app"
-  topics:
-  - detection in place
-  - Effectiveness Assessment (e.g. bypassing the detection)
+  masvs-v1: [MSTG-RESILIENCE-3, MSTG-RESILIENCE-4, MSTG-RESILIENCE-5, MSTG-RESILIENCE-6]
+  masvs-v2: [MASVS-RESILIENCE-1, MASVS-RESILIENCE-4]
```

Comment:
Are there examples of app virtualization being used to dynamically analyse an application? It's possible, I guess, but why would you go this route vs just repackaging the app in the first place and add a debugger / frida / custom smali?

If not, we should remove resilience-4

Reply:
It's an alternative way. Many application protection solutions provide root detection, anti-repackaging, and dynamic instrumentation (Frida,  LSPosed, etc.) detection guards but they can't effectively detect virtualization environments. Virtualization doesn't require root, most virtualizers don't tamper APK integrity in any way - two fewer checks to worry about for the attacker.

Of course, it's not as efficient as using Frida on a rooted device but virtualization software provides a lot of its benefits as a trade-off.

Reply:
Technically I can't immediately see how the virtualizers *don't* tamper with APK integrity. For cloning, the signature is definitely changed; for anything that wraps an APK, the signature is also changed (but calls to system APIs are hooked to return original signature, which is similar to hooking those things with Frida/xposed or smali patching).

It's ok if we keep this, but it would be nice to have an actual example of this technique being used for dynamic analysis. Maybe gameguardian could be enough (dynamically creating cheats or disabling ssl pinning or intercepting requests?) but I have limited experience with gameguardian.

Reply:
> Technically I can't immediately see how the virtualizers don't tamper with APK integrity. For cloning, the signature is definitely changed; for anything that wraps an APK, the signature is also changed (but calls to system APIs are hooked to return original signature, which is similar to hooking those things with Frida/xposed or smali patching).

It really depends on virtualization software. Some tamper, some don't. I think everything comes down to writing actual MASTG tests and adding demos here. I was planning to do that in the next iterations.